index_batcher.go

```diff
@@ -0,0 +1,164 @@
+package bleve
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NOTE XXX As this approach uses carefully orchestrated interactions
+// between a mutex, channels, and reference swapping, be sure to carefully
+// consider the impact of reordering statements or making changes to the
+// patterns present in the code.
+
+// bleveIndex is an alias that allows the IndexBatcher to embed a Index
+type bleveIndex Index
```

Comment:
Yeah I gave up to quickly on trying to figure out how to make Go dynamically allocate a nil error interface (turns out new(error)) and instead thought "well I'll just wrap a concrete type around the interface since its easier to work with a pointer to a concrete type than a pointer to an interface anyway".  Your comment drove me to fix it though.